vhost/src/vhost_user/message.rs

```diff
@@ -987,6 +991,45 @@ impl VhostUserMsgValidator for VhostUserTransferDeviceState {
     }
 }
 
+// Bit mask for flags in VhostUserMMap struct
+bitflags! {
+    #[derive(Copy, Clone, Debug, Eq, PartialEq, Default)]
+    /// Flags specifying access permissions of memory mapping of a file
+    /// Use VhostUserMMapFlags::default() for a read-only mapping
+    pub struct VhostUserMMapFlags: u64 {
+        /// Read-write permission
+        const WRITABLE = 1 << 0;
+    }
+}
+
+/// Backend request to mmap a file-backed buffer into guest memory
+#[repr(C, packed)]
+#[derive(Debug, Copy, Clone, Default)]
+pub struct VhostUserMMap {
+    /// Shared memory region ID.
+    pub shmid: u8,
+    /// Struct padding.
+    pub padding: [u8; 7],
+    /// File offset.
+    pub fd_offset: u64,
+    /// Offset into the shared memory region.
+    pub shm_offset: u64,
+    /// Size of region to map.
+    pub len: u64,
+    /// Flags for the mmap operation
+    pub flags: u64,
+}
+
+// SAFETY: Safe because all fields of VhostUserMMap are POD.
+unsafe impl ByteValued for VhostUserMMap {}
+
+impl VhostUserMsgValidator for VhostUserMMap {
```

Comment:
Should we add a test for this, as we do for other `VhostUserMsgValidator`  implementations (e.g. `check_user_memory()`, `check_user_memory_region()`, etc.)